src/utils/db.ts

```diff
@@ -295,17 +314,21 @@ export class AuctioneerDatabase {
   }
 
   /**
-   * Get all users in the database with a collateral position for the given asset.
+   * Get all users from `pool_id` in the database with a collateral position for the given asset.
+   * @param poolId - The pool id the user belongs to
    * @param assetId - The asset to filter by
    * @returns An array user entries, or an empty array if none are found
    */
-  getUserEntriesWithCollateral(assetId: string): UserEntry[] {
+  getUserEntriesWithCollateral(poolId: string, assetId: string): UserEntry[] {
     try {
       let entries: any[] = this.db
-        .prepare('SELECT * FROM users WHERE json_extract(collateral, ?) IS NOT NULL')
-        .all(`$.value.${assetId}`);
+        .prepare(
+          'SELECT * FROM users WHERE json_extract(collateral, ?) IS NOT NULL AND pool_id = ?'
```

Comment:
we should put the `pool_id` portion of the query first, given it's part of the primary key so sqlite should (I think?) make use of the index to prefilter before `json_extract` is used